File: core/fitness/trainings/TrainingDTO.ts
import { TrainingTaxonomy } from "../TrainingTaxonomy";

export interface TrainingVideoDTO {
    name: string
    ext: string
    length: number
}

export interface TrainingDTO {
    categories: TrainingTaxonomy[]
    trainerId: string
    name: string
    description: string
    trainingVideo: TrainingVideoDTO
}

File: fitness-dimension/application/trainings/training_dto.go
package trainings

type TrainingDto struct {
	ID          string            `json:"id"`
	Name        string            `json:"name"`
	TrainerID   string            `json:"trainer_id"`
	Description string            `json:"description"`
	Categories  []string          `json:"categories"`
	Video       *TrainingVideoDto `json:"video"`
}

type TrainingVideoDto struct {
	ID     string `json:"id"`
	Name   string `json:"name"`
	Ext    string `json:"ext"`
	Buff   []byte `json:"buff"`
	Length int    `json:"length"`
}
